src/main/webapp/js/user-page-loader.js

```diff
@@ -100,14 +100,14 @@ function buildMessageDiv(message) {
   headerDiv.appendChild(document.createTextNode(my_message));
   const bodyDiv = document.createElement('div');
   bodyDiv.classList.add('message-body');
-  var out_text = message.text.split(" ");
-  for (var i = 0; i < out_text.length; i++) {
-    if (out_text[i].substring(0,1) === "@")
-    {
-      out_text[i] = out_text[i].fontcolor("red");
-    }
+  var regex = /@(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+  var found = regex.exec(message.text)
+  while (found != null) {
+    var foundEmail = found[0].substring(1);
+    message.text = message.text.replace(found[0], foundEmail.fontcolor("red").link("/user-page.html?user=" + foundEmail));
```

Comment:
Currently set to make font color red; @pabloemilio97 might know a good color/setting for mentions?